snap/snapcraft.yaml

```
@@ -13,16 +13,17 @@ description:
     A download from GOG.com or Origin
   https://github.com/CorsixTH/CorsixTH/wiki/Getting-Started
 adopt-info: corsixth
-base: core18
+base: core22
 
 grade: stable
 confinement: strict
 
 apps:
   corsixth:
-    command: desktop-launch $SNAP/bin/corsix-th $SNAP/share/corsix-th/CorsixTH.lua
+    extensions: [ gnome ]
+    command: bin/launch
```

Have not tested, but this should work, removing the need for `snap/local/bin/launch`
```suggestion
    command: usr/local/bin/corsix-th --interpreter=$SNAP/usr/local/share/corsix-th/CorsixTH.lua
```

---

I tried that and snapcraft complained about erroneous characters in the command line, so I moved it out to a launch script.